### PR TITLE
make canister log public

### PR DESCRIPTION
**Pull request by chenyan-dfinity**

### CODE REVIEW

File: service/pool/IC.mo [hunk 1]
```diff
@@ -2,19 +2,22 @@
 module {
   public type canister_id = Principal;
   public type snapshot_id = Blob;
+  public type log_visibility = { #controllers; #public_ };
```
Comment (ggreif) [inline service/pool/IC.mo:5]: `#everyone`?

File: service/pool/Main.mo [hunk 2]
```diff
@@ -149,6 +157,7 @@ shared (creator) actor class Self(opt_params : ?Types.InitParams) = this {
             pool.removeCanister(info);
             let settings = {
                 controllers = ?controllers;
+                log_visibility = ?(#controllers);
```
Comment (ggreif) [inline service/pool/Main.mo:160]: The parens are redundant. In other places too.

File: service/pool/IC.mo (hunk 1 above)
Comment (ggreif) [reply service/pool/IC.mo:5]: `#unrestricted`?

Comment (ggreif) [reply service/pool/IC.mo:5]: Ah, maybe this variant is dictated by the management canister and we have no choice?

Comment (chenyan-dfinity) [reply service/pool/IC.mo:5]: Yep, that's from the spec.

**Review by rvanasa — APPROVED** [review]

**Pull request merged**